gui/src/components/Balances.tsx

```diff
@@ -16,20 +16,23 @@ export function Balances() {
     "db_get_erc20_balances",
     { address }
   );
+  const { data: settings } = useInvoke<GeneralSettings>("settings_get");
+
+  const reorderedBalances = (balances || [])
+    .map<[Address, bigint]>(([c, b]) => [c, BigInt(b)])
+    .filter(([, balance]) => (settings?.hideEmptyTokens ? !!balance : true));
 
   useRefreshTransactions(mutate);
 
   return (
     <Panel>
       <Stack>
         {address && <BalanceETH address={address} />}
-        {(balances || []).map(([contract, balance]) => (
+        {reorderedBalances.map(([contract, balance]) => (
           <BalanceERC20
             key={contract}
-            {...{
-              contract,
-              balance: BigInt(balance),
-            }}
+            contract={contract}
+            balance={BigInt(balance)}
```

Comment:
this can now be just `balance={balance}` right?

Reply:
yap, my bad